// statik/lib/statik.js
var path = require('path');
var extend = require('extend');
var connect = require('connect');
var serveStatic = require('serve-static');
var modRewrite = require('connect-modrewrite');
var redirect = require('connect-redirection');

var defaults = {
  /* connect options */
  port: 3000,

  /* connect.static options */
  root: __dirname,
  maxAge: 0,
  hidden: false,
  redirect: true,
  compress: true
};

module.exports = function (opts) {
  var options = extend(Object.create(null), defaults, opts);

  // setup middlewares
  //var compress = connect.compress();
  var static = serveStatic(options.root,{'index': ['index.html']});

  // start the app on given port. defaults to 3000
  var app = connect();

  /*app.use(redirect());
  app.use(function(req, res, next) {
    if(req.headers['x-forwarded-proto']!='https') {
      return res.redirect(['https://app.releaseboard.com', req.url].join(''));
    } else {
      next();
    }
  });*/
  app.use(modRewrite(['^[^\\.]*$ /index.html [L]']));
  //app.use(modRewrite(['!\\.html|\\.js|\\.css|\\.jpeg|\\.svg|\\.gif|\\.ttf|\\.woff|\\.eot|\\.jpg|\\.png$ /index.html [L]']));

  //app.use(compress);
  app.use(static);
  console.log('Server listening on port '+options.port+', ctrl+c to cancel.');
  app.listen(options.port);
  return app;
};
